src/TorchSharp/Data/Dataset.cs

```diff
@@ -0,0 +1,14 @@
+using System;
+using System.Collections.Generic;
+
+using static TorchSharp.torch;
+
+namespace TorchSharp.Data
+{
+    public interface Dataset: IDisposable
+    {
+        public long Count();
+        public (Tensor, Tensor) GetTensor(object index);
+        public IEnumerable<object> GetDataEnumerable();
```

Comment:
remove "public IEnumerable<object> GetDataEnumerable();" this should not be part of interface